actors/builtin/miner/miner_test.go

```diff
@@ -2052,15 +2052,129 @@ func TestWithdrawBalance(t *testing.T) {
 
 func TestChangePeerID(t *testing.T) {
 	periodOffset := abi.ChainEpoch(100)
+	actor := newHarness(t, periodOffset)
+	builder := builderForHarness(actor).
+		WithBalance(bigBalance, big.Zero())
 
 	t.Run("successfully change peer id", func(t *testing.T) {
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		newPID := tutil.MakePID("test-change-peer-id")
+		actor.changePeerID(rt, newPID)
+	})
+}
+
+func TestCompactPartitions(t *testing.T) {
```

Comment:
CompactPartitions garbage collects terminated sectors. Specifically, it:

1. Deletes the specified partitions.
2. Sorts the sectors into a live set and a dead set (terminated).
3. _Deletes_ the sector infos for the dead set.
4. Re-adds the live set back to the deadline they came from.

Tests here should test:

* We cannot compact partitions with faults.
* We cannot compact partitions with pending early terminations (e.g., a fault expired and the fee hasn't yet been payed). This may be a bit difficult to test.
* If we compact a partition with 5 live sectors and 5 dead, the 5 dead should be deleted (from the state's Sectors array) and the 5 live should be still be present (check with `FindSector`) in the deadline at which they started.

Reply:
@Stebalien I've added tests for **1** AND **3** but am running into this problem:

https://github.com/filecoin-project/specs-actors/pull/969#discussion_r471251481.